src/tests/unit/tests/reports/components/new-tab-link-confirmation-dialog.test.tsx

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { NewTabLink } from 'common/components/new-tab-link';
+import { shallow } from 'enzyme';
+import { isFunction } from 'lodash';
+import * as React from 'react';
+import {
+    ConfirmType,
+    NewTabLinkWithConfirmationDialog,
+} from 'reports/components/new-tab-link-confirmation-dialog';
+import { IMock, It, Mock, MockBehavior, Times } from 'typemoq';
+
+describe('NewTabLinkWithConfirmationDialog', () => {
+    it('generates unique id for each link', () => {
+        const testSubject1 = shallow(<NewTabLinkWithConfirmationDialog />);
+        const testSubject2 = shallow(<NewTabLinkWithConfirmationDialog />);
+
+        expect(testSubject1.find(NewTabLink).prop('id')).not.toEqual(
+            testSubject2.find(NewTabLink).prop('id'),
+        );
+    });
+
+    it('contains NewTabLink with forwarded props', () => {
+        const props = {
+            href: 'test-href',
+            title: 'test-title',
+        };
+        const child = <div>test-child</div>;
+
+        const testSubject = shallow(
+            <NewTabLinkWithConfirmationDialog {...props}>{child}</NewTabLinkWithConfirmationDialog>,
+        );
+
+        const testSubjectNewTabLink = testSubject.find(NewTabLink);
+
+        expect(testSubjectNewTabLink.prop('href')).toEqual(props.href);
+        expect(testSubjectNewTabLink.prop('title')).toEqual(props.title);
+        expect(testSubjectNewTabLink.contains(child)).toBe(true);
+    });
+
+    describe('onClickHandler', () => {
+        const targetPageLinkMock = Mock.ofType<HTMLElement>();
+
+        type GetElementById = typeof Document.prototype.getElementById;
+
+        let getElementByIdMock: IMock<GetElementById>;
+        let confirmMock: IMock<ConfirmType>;
+
+        let originalGetElementById: GetElementById;
+        let originalConfirm: ConfirmType;
+
+        beforeEach(() => {
+            originalGetElementById = document.getElementById;
+            originalConfirm = window.confirm;
+
+            getElementByIdMock = Mock.ofType<GetElementById>(undefined, MockBehavior.Strict);
+            confirmMock = Mock.ofType<ConfirmType>(undefined, MockBehavior.Strict);
+
+            getElementByIdMock
+                .setup(handler => handler(It.isAnyString()))
+                .returns(() => targetPageLinkMock.object);
+
+            document.getElementById = getElementByIdMock.object;
+            window.confirm = confirmMock.object;
+        });
+
+        afterEach(() => {
+            document.getElementById = originalGetElementById;
+            window.confirm = originalConfirm;
+        });
+
+        it('is added to the link', () => {
+            getElementByIdMock.reset();
```

Comment:
is this call to `.reset()` necessary here? I see on line 56 we are creating a new instance before each test:
```ts
getElementByIdMock = Mock.ofType<GetElementById>(undefined, MockBehavior.Strict);
```

Reply:
In line 59 (in `beforeEach`), we set up `getElementIdMock` in a more general way, without specifying the id. Do you think it would be better to move the setup to each individual test, so we don't have to reset the mock setup here? Or should we set up the mock such that we use the specific id each time?

Reply:
so if that setup on line 59 does not apply to all the tests under this `describe` maybe it should not be on the `beforeEach` ??

Reply:
Moved setup out of beforeEach